fairlearn/grid_search/binary_protected_attribute/regression.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import copy
+import numpy as np
+
+from . import utilities
+
+
+def _regression_weight_function(a_val, trade_off, p0, p1, a0_val):
+    if a_val == a0_val:
+        return trade_off / p0
+    else:
+        return (1 - trade_off) / p1
+
+
+def bounded_group_loss(learner,
+                       x, y, protected_attribute,
+                       tradeoffs=None,
+                       number_of_tradeoffs=11):
+    # Must specify either an array of tradeoffs or the number
+    # of tradeoffs to generated
+    if not (tradeoffs is None) ^ (number_of_tradeoffs is None):
```

Comment:
please use the pythonic way of expressing this
if tradeoffs is not None and number_of_tradeoffs is None:
```

Reply:
That's not what I'm expressing. As the comment says this is an either/or. According to what I've read, `^` is XOR ... and then I do the `not` of that

Reply:
Changed to way we agreed on teams